packages/hint-compat-api/src/css-base.ts

```diff
@@ -0,0 +1,160 @@
+    private readonly statusName: CSSFeatureStatus;
+    private compatApi: CompatApi;
+    private compatCSS: CompatCSS;
+
+    abstract getFeatureVersionValueToAnalyze(browserFeatureSupported: SimpleSupportStatement): VersionValue;
+    abstract isSupportedVersion(browser: BrowsersInfo, feature: FeatureInfo, currentVersion: number, version: number): boolean;
+    abstract isVersionValueSupported(version: VersionValue): boolean;
+    abstract isVersionValueTestable(version: VersionValue): boolean;
+
+    public constructor(context: HintContext<StyleEvents>, statusName: CSSFeatureStatus, isCheckingNotBroadlySupported: boolean) {
+        this.compatApi = new CompatApi('css', context, isCheckingNotBroadlySupported);
+        this.compatCSS = new CompatCSS(context, this.testFeatureIsSupported.bind(this));
+        this.statusName = statusName;
+
+        context.on('parse::end::css', this.onParseCSS.bind(this));
+    }
+
+    private async onParseCSS(styleParse: StyleParse): Promise<void> {
+        const { resource } = styleParse;
+
+        this.compatCSS.setResource(resource);
+        await this.compatCSS.searchCSSFeatures(this.compatApi.compatDataApi, styleParse);
+    }
+
+    private async testFeatureIsSupported(feature: FeatureInfo, supportBlock: SupportBlock): Promise<void> {
+        const browsersToSupport = Object.entries(supportBlock).filter(([browserName, _]: [string, SupportStatement]): boolean => {
+            return this.compatApi.isBrowserIncludedInCollection(browserName);
+        });
+
+        const groupedSupportByBrowser = browsersToSupport.reduce((group, browserInfo, _, a1) => {
+            return this.groupSupportStatementByBrowser(feature, group, browserInfo);
+        }, {});
+
+        const supportStatementResult: SupportStatementResult = {
+            browsersToSupportCount: browsersToSupport.length,
+            groupedBrowserSupport: groupedSupportByBrowser,
+            notSupportedBrowsersCount: Object.keys(groupedSupportByBrowser).length
+        };
+
+        const hasImcompatibleBrowsers: boolean = supportStatementResult.notSupportedBrowsersCount > 0;
+
+        if (!hasImcompatibleBrowsers) {
+            return;
+        }
+
+        const message: string = this.generateReportErrorMessage(feature, supportStatementResult);
+
+        await this.compatCSS.reportError(feature, message);
+    }
+
+    private groupSupportStatementByBrowser(feature: FeatureInfo, group: { [browserName: string]: string[] }, browserInfo: [string, SupportStatement]) {
+        const [browserName, supportStatement] = browserInfo;
+        const browserFeature = this.compatApi.getSupportStatementFromInfo(supportStatement, feature.prefix);
+        const info: BrowsersInfo = { browserInfo: supportStatement, browserToSupportName: browserName };
+        const versions = browserFeature && this.getNotSupportedBrowser(info, feature, browserFeature);
+
+        if (versions) {
```

Comment:
Not super important and maybe we did the same in the past but usually seems more legible to have something like

```
if (!versions) {
return group;
}

return {..group........}
```

Reply:
ok, I don't get why this is more legible than the previous approach :P.

Reply:
In any case I can update my code ;)

Reply:
probably is not an objective comment xD if you have 2 or 3 ifs to check this kind of things, usually is more readble to read if not go out, if not go out, if not go out, everything is ok, return x